src/parser.rs

```diff
@@ -2180,3 +2171,83 @@ where
         self.fix_position(err)
     }
 }
+
+impl<'de, R> Parser<R>
+where
+    R: Reader<'de>,
+{
+    pub fn get_by_schema(&mut self, schema: &mut crate::Value) -> Result<()> {
+        if !schema.is_object() {
+            return perr!(
+                self,
+                Message(std::borrow::Cow::Borrowed("The schema must be an object"))
+            );
+        }
+
+        let mut strbuf = Vec::with_capacity(DEFAULT_KEY_BUF_CAPACITY);
+        self.get_by_schema_rec(schema, &mut strbuf)
+    }
+
+    fn get_by_schema_rec(&mut self, schema: &mut crate::Value, strbuf: &mut Vec<u8>) -> Result<()> {
+        let ch = self.skip_space_peek();
+        if ch.is_none() {
+            return perr!(self, EofWhileParsing);
+        }
+
+        let mut should_replace = true;
+        let start = self.read.index();
+
+        match (schema.as_object_mut(), ch) {
+            (Some(object), Some(b'{')) => {
+                let mut key_values = HashMap::new();
+                for (key, value) in object.iter_mut() {
+                    key_values.insert(key, value);
+                }
+
+                // We should replace the schema object if the object is empty
+                should_replace = key_values.is_empty();
+                if should_replace {
+                    self.skip_one()?;
+                } else {
+                    // deal with the empty object
+                    match self.get_next_token([b'"', b'}'], 1) {
```

Comment:
The API is not marked unsafe, so we should always validate the remaining parts. maybe json likes `{123"key":`

Reply:
I think it has been validated by https://github.com/cloudwego/sonic-rs/pull/110/files/416f090ee57e6e84f482eb09272b3d71cf20eca3#diff-4a04259da480a6b794a2e947e4cc03eff4d1aa9330836f5b91cac68c5398193fR2229, which I copy from `get_many_keys`

Reply:
i see, i think this is also a bug in `get_many_keys`...